src/sentry/api/endpoints/organization_events.py

```diff
@@ -218,64 +232,155 @@ def get(self, request, organization):
             }, status=400)
 
         try:
-            tag_keys = tagstore.get_group_tag_keys_and_top_values(
-                project_ids, None, environment_ids, keys=lookup_keys, get_excluded_tags=True, **snuba_args)
-        except tagstore.TagKeyNotFound:
-            raise ResourceDoesNotExist
+            total_count = self.get_total_value_count(**snuba_args)
+        except (KeyError, SnubaError):
+            logger.info(  # TODO(lb): should this be level error?
+                'api.organization-events-heatmap',
+                extra={
+                    'organization_id': organization.id,
+                    'user_id': request.user.id,
+                    'keys': keys,
+                    'snuba_args': snuba_args,
+                }
+            )
+            return Response({
+                'detail': 'Invalid query.'  # TODO(lb): is this ok?
```

Comment:
If we have more context that we can give to the user that would be helpful for them in solving the problem it would be good to include that here.